### PR TITLE
Wwarren 813 health stat annot

### PR DESCRIPTION
**Pull request by wildlyadequate**

### CODE REVIEW

File: src/main/java/com/expedia/seiso/domain/repo/impl/AnnotatedHealthStateRepoImpl.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+
+import com.expedia.seiso.domain.repo.custom.AnnotatedHealthStatusRepoCustom;
+import com.expedia.seiso.web.resource.AnnotatedHealthStatus;
+import com.expedia.seiso.web.resource.NodeSummary;
+
+import lombok.val;
+
+public class AnnotatedHealthStateRepoImpl implements AnnotatedHealthStatusRepoCustom {
```
Comment (williewheeler) [inline src/main/java/com/expedia/seiso/domain/repo/impl/AnnotatedHealthStateRepoImpl.java]: Change to AnnotatedHealthStatusRepoImpl

File: src/main/java/com/expedia/seiso/web/controller/AnnotatedHealthStatusController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import com.expedia.seiso.web.assembler.AnnotatedHealthStatusService;
+import com.expedia.seiso.web.assembler.EnvironmentService;
+import com.expedia.seiso.web.resource.AnnotatedHealthStatus;
+import com.expedia.seiso.web.resource.ServiceInstanceResource;
+
+@BasePathAwareController
+@RequestMapping("/healthStatus/")
```
Comment (williewheeler) [inline src/main/java/com/expedia/seiso/web/controller/AnnotatedHealthStatusController.java]: The repo path should be pluralized like the rest of the API. But there is already a healthStatuses repo. (Still looking around to better understand the changes.)

**Comment by williewheeler** [comment]
Let's get together to discuss this in person, but I want to suggest a different approach:

1. Create an actual AnnotatedHealthStatus entity. (Not just a resource. Indeed I don't think we need the resource at all.) The AnnotatedHealthStatus entity would have a HealthStatus and a log link. (So remove the log link from the Node.)
2. Replace the Node's HealthStatus with an AnnotatedHealthStatus.
3. Don't use custom controllers or repos for any of this. At least I don't see why we need to if we adopt #1 and #2 above.

Conceptually the log link attaches to the health status, so that's why I am thinking we want to bind that to the HealthStatus more explicitly. For example if we were to add a log link for rotation status (which in fact was a request--it would explain why the node was pulled out of rotation), then that would compete with the existing logLink property, or we'd have to rename it to healthStatusLogLink vs rotationStatusLogLink, etc. I think cleaner simply to bind the explanation to the status itself.

Like I said, let's discuss in person.

**Comment by wildlyadequate** [comment]
OK, I can be back there tomorrow afternoon, does that work?

**Comment by williewheeler** [comment]
Sure thing.

**Pull request closed without merging**